python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,799 @@
+        ), "Invalid number of nodes"
+        assert (
+            len(self._data[STR_NUM_EDGES_PER_TYPE]) > 0
+        ), "Invalid number of edges"
+
+        assert (
+            len(self._data[STR_NUM_NODES_PER_TYPE]) > 0
+        ), "Invalid number of nodes"
+        assert (
+            len(self._data[STR_NUM_EDGES_PER_TYPE]) > 0
+        ), "Invalid number of edges"
+
+        for idx, num_nodes in enumerate(
+            self._data[STR_NUM_NODES_PER_TYPE]
+        ):
+            if num_nodes > 0:
+                continue
+            raise ValueError(
+                f"No. of nodes at index: {idx} is not a valid value."
+            )
+
+        for idx, num_edges in enumerate(
+            self._data[STR_NUM_EDGES_PER_TYPE]
+        ):
+            if num_edges > 0:
+                continue
+            raise ValueError(
+                f"No. of edges at index: {idx} is not a valie value."
+            )
+
+        assert len(self._data[STR_NODE_TYPE]) == len(
+            self._data[STR_NUM_NODES_PER_TYPE]
+        ), "No. of node types does not match with No. of nodes per node type."
+        assert len(self._data[STR_EDGE_TYPE]) == len(
+            self._data[STR_NUM_EDGES_PER_TYPE]
+        ), "No. of edge types does not match with No. of edges per edge type."
+
+        assert len(self._ntype_features) <= len(
+            self._data[STR_NUM_NODES_PER_TYPE]
+        ), "Node types with features does not match node type counts."
+        assert len(self._etype_features) <= len(
+            self._data[STR_NUM_EDGES_PER_TYPE]
+        ), "Edge types with features does not match edge type counts."
+
+        # Check for each existing node feature, corresponding node type is valid
+        for key, _ in self._ntype_feature_files.items():
+            ntype = key[0]
+            if ntype in self.ntypes:
+                continue
+            raise ValueError(
+                f"Node Type: {ntype} is not present in the list of node types."
+            )
+
+        # Check for each existing edge feature, corresponding edge type is valid
+        for key, _ in self._etype_feature_files.items():
+            etype = key[0]
+            if etype in self.etypes:
+                continue
+            raise ValueError(
+                f"Edge Type: {etype} is not present in the list of edge types"
+            )
+
+    def __del__(self):
+        self._basedir = None
+        self._metadata_path = None
+        self._data = None
+        self._global_nid_offsets = None
+        self._global_eid_offsets = None
+        self._ntypes = None
+        self._ntype_id_map = None
+        self._id_ntype_map = None
+        self._etype_id_map = None
+        self._id_etype_map = None
+        self._etypes = None
+        self._ntype_features = None
+        self._etype_features = None
+        self._ntype_feature_files = None
+        self._etype_feature_files = None
+        self._etype_files = None
```

Comment:
I'd be surprised if having a destructor adds value here. All these member vars will get released and gc'ed when the object goes out of scope right?

Reply:
Removed.